Adicionando diagrama de componentes e a descrição de cada componente. (#1)
RodrigoEC/arqsoft-blog: content/posts/hoco.md

O HoCo tem como objetivo principal sanar uma deficiência conhecida no curso de Ciência da Computação na UFCG, a falta de conhecimento sobre o funcionamento de horas e atividade complementares do curso.

O projeto visa criar uma plataforma onde os alunos poderão fazer o gerenciamento das suas atividades complementares de forma simples e unificada. Ao fazer uso do HoCo o aluno poderá cadastrar suas atividades complementares e estas serão enviadas para a coordenação do curso para que essas sejam avaliadas quanto a sua corretude e validez de forma automática.

## Diagrama de componetes

Abaixo, é possível observar o diagrama de componentes do sistema:

![ComponetDiagram](/content/posts/hoco/diagrama-componetes.png)

**Nos componentes, temos:**

*Os componentes referentes à sessão de perguntas e organizações são bastante semelhantes.*

> **Questions Controller e Organization Controller:** São REST Controllers do Flask que fazem o controle sobre a sessão de perguntas e de organização do HoCo, respectivamente, pois é possível que o usuário cadastre dúvida para que sejam respondidas e organizações no HoCo, esses componetes fazem uso respectivamente dos componetes Questions Modells e Organization Modells. Ambos não precisam de login;

Reviewer: ```suggestion
**Questions Controller e Organization Controller:** São REST Controllers do Flask que fazem o controle sobre a sessão de perguntas e de organização do HoCo, respectivamente, pois é possível que o usuário administrador cadastre dúvidas e suas respostas e organizações e comunidades existentes na graduação no HoCo, esses componetes fazem uso respectivamente dos componetes Questions Modells e Organization Modells. Ambos não precisam de login;
```

> **Questions Modells e Organization Modells:** São componetes que fazem a conexão com o Banco de Dados, fazendo leitura e/ou escrita dos dados no mesmo;

Reviewer: ```suggestion
**Questions Models e Organization Models:** São componetes que fazem a conexão com o Banco de Dados, fazendo leitura e/ou escrita dos dados no mesmo;
```

> **Login Controller:** Esse componente é um REST Controller do Flask que faz o controle da sessão de login, é responsável por realizar o login e logout do usuário e a restauração da senha do usuário, fazendo ususo do componente Security Component;

Reviewer: ```suggestion
**Login Controller:** Esse componente é um REST Controller do Flask que faz o controle da sessão de login, é responsável por realizar o login e logout do usuário e a restauração da senha do usuário, fazendo uso do componente Security Component;
```

**Security Component:** É o componente responsável por criptografar e descriptografar as senhas dos usuários, verificar se os dados de login estão corretos, verificar se o usuário tem permissão de acesso à funcionalidade solicitada pelo mesmo;

> **User Controller:** É um REST Controller do Flask que provê todas as funcionalidades referentes à um usuário em específico (por isso faz uso do Security Component, pois é necessário que o usuário esteja logada, exceto no cadastro), como o cadastro do usuário (não precisa de login), exclusão de conta, cadastro de atividades, exclusão de atividade, edição de perfil, aprovar atividade (precisa que o usuário tenha mais permissões, não permitindo que um aluno aprove atividades);

Reviewer: ```suggestion
**User Controller:** É um REST Controller do Flask que provê todas as funcionalidades referentes à um usuário em específico, por isso faz uso do Security Component, pois é necessário que o usuário esteja logada para que a pessoa possa acessar funcionalidades especificas, como cadastro de atividades (Rota de acessoa 
 dúvidas e respostas, organizações e cadastro do usuário são públicas para qualquer pessoa acessar sem login), como o cadastro do usuário (não precisa de login), exclusão de conta, cadastro de atividades, exclusão de atividade, edição de perfil, aprovar atividade (precisa que o usuário tenha mais permissões, não permitindo que um aluno aprove atividades);
```

> **User Modells:** É o componenente que faz a conexão com o Banco de Dados, fazendo a leitura e escrita dos dados no mesmo;

Reviewer: ```suggestion
**User Models:** É o componenente que faz a conexão com o Banco de Dados, fazendo a leitura e escrita dos dados no mesmo;
```

O HoCo tem como objetivo principal sanar uma deficiência conhecida no curso de Ciência da Computação na UFCG, a falta de conhecimento sobre o funcionamento de horas e atividade complementares do curso.

O projeto visa criar uma plataforma onde os alunos poderão fazer o gerenciamento das suas atividades complementares de forma simples e unificada. Ao fazer uso do HoCo o aluno poderá cadastrar suas atividades complementares e estas serão enviadas para a coordenação do curso para que essas sejam avaliadas quanto a sua corretude e validez de forma automática.

## Diagrama de componetes

Abaixo, é possível observar o diagrama de componentes do sistema:

![ComponetDiagram](/content/posts/hoco/diagrama-componetes.png)

**Nos componentes, temos:**

*Os componentes referentes à sessão de perguntas e organizações são bastante semelhantes.*

> **Questions Controller e Organization Controller:** São REST Controllers do Flask que fazem o controle sobre a sessão de perguntas e de organização do HoCo, respectivamente, pois é possível que o usuário administrador cadastre dúvidas e suas respostas e organizações e comunidades existentes na graduação no HoCo, esses componetes fazem uso respectivamente dos componetes Questions Modells e Organization Modells. Ambos não precisam de login;

Reviewer 2: alterar componetes para **componentes**

> **Questions Models e Organization Models:** São componetes que fazem a conexão com o Banco de Dados, fazendo leitura e/ou escrita dos dados no mesmo;

Reviewer 2: alterar componetes para componentes

**Security Component:** É o componente responsável por criptografar e descriptografar as senhas dos usuários, verificar se os dados de login estão corretos, verificar se o usuário tem permissão de acesso à funcionalidade solicitada pelo mesmo;

**User Controller:** É um REST Controller do Flask que provê todas as funcionalidades referentes à um usuário em específico, por isso faz uso do Security Component, pois é necessário que o usuário esteja logada para que a pessoa possa acessar funcionalidades especificas, como cadastro de atividades (Rota de acessoa 
 dúvidas e respostas, organizações e cadastro do usuário são públicas para qualquer pessoa acessar sem login), como o cadastro do usuário (não precisa de login), exclusão de conta, cadastro de atividades, exclusão de atividade, edição de perfil, aprovar atividade (precisa que o usuário tenha mais permissões, não permitindo que um aluno aprove atividades);

**User Models:** É o componenente que faz a conexão com o Banco de Dados, fazendo a leitura e escrita dos dados no mesmo;

> **Atividade Service:** É o componente responsável por prover todas as funcionalidades referentes às atividades, como cadastro, cálculo de créditos, recuperação das atividades e suas informações. Esse componente tambem faz conexão com o Banco de Dados, fazendo a leitura e escrita dos dados no mesmo;

Reviewer 2: alterar tambem por **também**
